docs/source/upgrade/upgrade-guide-from-monitoring-3.x-to-monitoring-4.y.rst

```diff
@@ -0,0 +1,266 @@
+are not present data will still be shown.
+
+Scylla Monitoring versions 4.0 and newer will rely only on recording rules.
+
+    .. note::
+       If you upgrade from a version older than 3.8 without back-filling, latency historical data will not be shown.
+
+The following instructions are only relevant if you are upgrading from a version older than 3.8 to a version 4.0 or higher,
+or if you are looking at historical data collected before you upgraded to 3.8.
+
+For example, you keep your data (retention period) for a year, and you upgraded to version 3.8 three months ago.
+
+In this example you have recording rules data only for the last three months, to be able to look at older latency
+information you will need to back-fill that missing period.
+
+The following instructions are based on the recording rules backfilling section in the `Prometheus documentation`_.
+
+.. _`Prometheus documentation` : https://prometheus.io/docs/prometheus/latest/storage/#backfilling-for-recording-rules
+
+Determine the backfilling period
+--------------------------------
+When you run the backfilling process you need to determine the start time and end time.
+
+Determine the start time
+^^^^^^^^^^^^^^^^^^^^^^^^
+The start time is your Prometheus retention time, by default it is set to 15 days. if you are not sure, you can check by
+login into your Prometheus server: `http://{ip}:9090/status`.
+If you are running Scylla Monitoring version 3.8 or newer for longer than the retention period, you are done! skip the rest of this section.
+
+For the rest of this example, we will assume that your retention time is 360 days.
+
+Determine the end time
+^^^^^^^^^^^^^^^^^^^^^^
+Typically, you need to back-fill the recording rules when you are using a long retention period, for example, you have a year of retention data,
+and you upgraded to Scylla Monitoring 3.8 about three months ago.
+
+If you open the Overview dashboard and look at your entire retention time (in our example 1 year) you will see that while most of the graphs do
+show the data, the latency graphs have a missing period, in our example - from the entire year, the latency graph will only show the last three months.
+
+.. figure:: before-backfilling.png
+   :width: 400pt
+
+   This is an example of missing latency graph
+
+That nine months gap (12 months minus 3) is what we want to fill with back-filling.
+
+The point in time that the graphs start will be your back-filling end time. Check in the graph for the exact time.
+
+Backfilling Process
+-------------------
+backup
+^^^^^^
+If you have a long retention period you are using an external directory that holds the Prometheus data, back it up, in case
+something would go wrong in the process you can revert the process.
+
+To complete the process you will need to restart the monitoring stack at least once, if you are not using an external directory (The ``-d``
+command-line option) You cannot complete it.
```

Comment:
Something's wrong with this sentence, but I'm not sure how to fix it. Perhaps it should be:
To complete the process you will need to restart the monitoring stack at least once. If you are not using an external directory (the ``-d``
command-line option), you cannot complete it.